src/scxml_sm_interface.cpp

```diff
@@ -136,6 +143,43 @@ ScxmlSMInterface::ScxmlSMInterface(const std::string& scxml_file)
   }
 }
 
+// use this to determine the next state in the state machine, given the name of the transition you'd like to query
+const QString ScxmlSMInterface::getNeighbor(const QString& state, const QString& search_text)
+{
+  QString next_state;
+
+  for (auto& pair : state_transition_map_.at(state))
+  {
+    if (pair.first == search_text)
+    {
+      return pair.second;
+    }
+  }
+  return next_state;
```

Comment:
I think I would prefer that this throws an exception if the state doesn't exist rather than returning an empty string

```suggestion
  for (auto& pair : state_transition_map_.at(state))
  {
    if (pair.first == search_text)
    {
      return pair.second;
    }
  }
  throw std::runtime_error("State '" + state.toStdString() + "' does not have a neighbor after transition '" + transition.toStdString() + "');
```

Reply:
@marip8  we have it as returning an empty string to trigger the button off functionality.
currently, we check that if the string it returns is empty, we turn off the button. 
It calls this function every state to determine if there's a nearby user action.